core/domain/exp_services.py

```diff
@@ -1970,6 +2050,62 @@ def update_exploration(
             message starts with the same prefix as the commit message for
             accepted suggestions.
     """
+    models_to_put = compute_models_for_updating_exploration(
+        committer_id=committer_id,
+        exploration_id=exploration_id,
+        change_list=change_list,
+        commit_message=commit_message,
+        is_suggestion=is_suggestion,
+        is_by_voice_artist=is_by_voice_artist,
+        is_synchronous=is_synchronous
+    )
+
+    datastore_services.update_timestamps_multi(models_to_put)
+    datastore_services.put_multi(models_to_put)
+
+
+def compute_models_for_updating_exploration(
+    committer_id: str,
+    exploration_id: str,
+    change_list: Optional[List[exp_domain.ExplorationChange]],
+    commit_message: Optional[str],
+    is_suggestion: bool = False,
+    is_by_voice_artist: bool = False,
+    is_synchronous: bool = False,
+) -> List[base_models.BaseModel]:
+    """Update an exploration. Commits changes.
```

Comment:
Does it actually commit the changes?

Please check all docstrings of all "compute model" functions. Make sure they are very clear about whether the datastore is being modified or not. This is important.